### ISSUE TITLE
networkChanged event called with different networkId to chainChanged event

### ISSUE THREAD

**Issue by aaronbarnardsound** [MetaMask/metamask-extension#13104]
**Describe the bug**
When registering for both the `networkChanged` and `chainChanged` events and switching between networks, the `chainChanged` listener will get called with one `chainId` and then the `networkChanged` listener will get called with a different `networkId`. This only seems to happen when switching to or from custom networks.

I know that the `networkChanged` event is deprecated. But in a project where we need to support all injected wallets, we need to register for the `networkChanged` event for backwards compatibility for wallets that do not yet support the `chainChanged` event.

**Steps to reproduce (REQUIRED)**
1. Add Avalanche and Matic as custom networks to MetaMask
2. Register listeners for both the `networkChanged` and `chainChanged` events and log the results
3. Start on the Matic network and then switch to Avalanche
4. You will see the `chainChanged` listener is called with `chainId`: `'0xa86a'`, then the `networkChanged` listener will be called with `'1'`

**Comment by Mdrafiqul65** [comment]
Network Binance Smart Chain. Contract address : 0x129385C4ACd0075e45A0C9a5177BdfEc9678A138
Name: Metakings Symbol MTK and Decimal 18

**Comment by Mdrafiqul65** [comment]
please address make me continue with wallet metamusk

**Comment by Gudahtt** [comment]
Thanks for the report! It would help if you shared the version of MetaMask you were using when you experienced this problem, along with the browser, browser version, and operating system that you are using.

**Comment by caishen7788** [comment]
can you help me also how to fix it i cant sell my token.. goin to broke now :(

**Comment by filipsekulic** [comment]
I did a research on this one and it seems to have been solved as it works properly.

Steps to check it:
  1. Go to test-dapp
  2. Connect your wallet
  3. Inspect 
  4. Console
  5. Enter: 
   - ethereum.on('chainChanged', (_chainId) => { console.log('CHAIN CHANGED: ' + _chainId) } ); 
   - ethereum.on(‘networkChanged', (_networkId) => { console.log(‘NETWORK CHANGED: ' + _networkId) } );
  6. Start on the Matic network and then switch to Avalanche
  
Screencast: 
https://user-images.githubusercontent.com/92310504/169288381-60173c21-7509-4711-8121-b8ec7af589d9.mov
